### PR TITLE
fix: fix `x/tokenfactory` genesis import denoms reset `x/bank` existing denom metadata

### PR DESCRIPTION
**Pull request by dadamu**
## What is the purpose of the change

This pull request fixes the `x/tokenfactory` genesis import reset `x/bank` existing denom metadata, say, `x/bank` genesis has the metadata of **facotory/osmos1.../testtoken** with `{Display: test}`, `x/tokenfactory` init genesis phase will replace it with the empty disply, more details as follows:

Before `x/tokenfactory` init genesis, the `x/bank` set metadata as:
```
{
  Name:        "Mint Token",
  Symbol:      "MTK",
  Description: "The custom token of the test subspace.",
  DenomUnits: []*banktypes.DenomUnit{
	{Denom: "factory/cosmos1y54exmx84cqtasvjnskf9f63djuuj68p7hqf47/uminttoken", Exponent: uint32(0), Aliases: nil},
	{Denom: "minttoken", Exponent: uint32(6), Aliases: []string{"minttoken"}},
  },
  Base:    "factory/cosmos1y54exmx84cqtasvjnskf9f63djuuj68p7hqf47/uminttoken",
  Display: "minttoken",
}
```

After `x/tokenfactory` init genesis, the metadata is replaced into:
```
{
  DenomUnits: []*banktypes.DenomUnit{
	{Denom: "factory/cosmos1y54exmx84cqtasvjnskf9f63djuuj68p7hqf47/uminttoken", Exponent: uint32(0)},
  },
  Base:    "factory/cosmos1y54exmx84cqtasvjnskf9f63djuuj68p7hqf47/uminttoken",
}
```

## Testing and Verifying

This change added tests and can be verified as follows:

* Added unit test that validates the `x/bank` imported metadata won't be reset by `x/tokenfactory`

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: x/tokenfactory/keeper/createdenom.go [hunk 1]
```diff
@@ -28,15 +28,18 @@ func (k Keeper) CreateDenom(ctx sdk.Context, creatorAddr string, subdenom string
...
+		denomMetaData := banktypes.Metadata{
+			DenomUnits: []*banktypes.DenomUnit{{
+				Denom:    denom,
+				Exponent: 0,
+			}},
+			Base: denom,
+		}
 
-	k.bankKeeper.SetDenomMetaData(ctx, denomMetaData)
+		k.bankKeeper.SetDenomMetaData(ctx, denomMetaData)
```
Comment (mattverse) [inline x/tokenfactory/keeper/createdenom.go:41]: Can unit tests for `CreateDenom` be added to confirm the new logic introduced as well?

**Review by mattverse — COMMENTED** [review]
Thanks for the PR! Logic change lgtm, requesting some test additions before we can merge this

**Comment by mattverse** [comment]
@dadamu Would you also be able to add CHANGELOG entry?

File: x/tokenfactory/keeper/createdenom.go (hunk 1 above)
Comment (dadamu) [reply x/tokenfactory/keeper/createdenom.go:41]: Added the unit test to check `CreateDenom` if the metadata is set correctly, the metadata existing case is already covered by [this test](https://github.com/osmosis-labs/osmosis/blob/main/x/tokenfactory/keeper/createdenom_test.go#L99)

**Comment by dadamu** [comment]
@mattverse Added the `CHANGELOG entry

**Review by mattverse — APPROVED** [review]
LGTM, thanks for the PR!

**Review by mattverse — APPROVED** [review]

**Pull request merged**